src/components/views/messages/MessageTimestamp.js

```diff
@@ -16,19 +16,28 @@ limitations under the License.
 
 'use strict';
 
-var React = require('react');
-var DateUtils = require('matrix-react-sdk/lib/DateUtils');
+import * as UserSettingsStore from 'matrix-react-sdk/lib/UserSettingsStore';
+const React = require('react');
+const DateUtils = require('matrix-react-sdk/lib/DateUtils');
 
 module.exports = React.createClass({
     displayName: 'MessageTimestamp',
 
     render: function() {
         var date = new Date(this.props.ts);
-        return (
-            <span className="mx_MessageTimestamp" title={ DateUtils.formatFullDate(date) }>
-                { DateUtils.formatTime(date) }
-            </span>
-        );
+        if (UserSettingsStore.getSyncedSetting('showTwelveHourTimestamps')) {
```

Comment:
technically the state of render() should depend only on props & state rather than js-sdk state, but i'm happy to ignore this for now :)